agent/app/agent.go

```diff
@@ -108,6 +110,26 @@ func newAgent(
 		return nil, err
 	}
 
+	var metadataManager containermetadata.MetadataManager
```

Comment:
I understand your purpose but I am reluctant to make this change because there are so many structs in the agent with the word Manager in them. Having a single struct called Manager, even if preceded by package containermetadata, feels confusing to me.